sql/updates/post-update.sql

```diff
@@ -30,3 +30,48 @@ $$;
 
 -- can only be dropped after views have been rebuilt
 DROP FUNCTION IF EXISTS _timescaledb_internal.cagg_watermark(oid);
+
+-- For objects that are newly created, we need to set the initprivs to
+-- the initprivs for the "chunk" catalog table since that is created
+-- in the first version of TimescaleDB and should have the correct
+-- initprivs even if the user as set the default privileges for the
+-- schema to something else.
+INSERT INTO saved_privs (
+     SELECT nspname, relname, relacl,
+ 	    (SELECT tmpini FROM saved_privs
+	      WHERE tmpnsp = '_timescaledb_catalog' AND tmpname = 'chunk')
+       FROM pg_class JOIN pg_namespace ns ON ns.oid = relnamespace
+		LEFT JOIN saved_privs ON tmpnsp = nspname AND tmpname = relname
+      WHERE nspname IN ('_timescaledb_catalog', '_timescaledb_config')
+ 	 OR nspname = '_timescaledb_internal'
+        AND relname IN ('hypertable_chunk_local_size', 'compressed_chunk_stats')
+) ON CONFLICT DO NOTHING;
+
+-- We can now update the initprivs for the objects using the saved
+-- privileges.
+-- 
+-- For objects that existed before, We need to restore initprivs to
```

Comment:
This comment is a bit confusing. Shouldn't initprivs for all the catalog objects be the same as that when the extension was first created? i believe that's what the sql does.

Reply:
Tried to clarify the comment.